tests/tracer/test_span.py

```diff
@@ -676,3 +676,19 @@ def test_manual_context_usage():
     span1.context.sampling_priority = 1
     assert span2.context.sampling_priority == 1
     assert span1.context.sampling_priority == 1
+
+
+def test_set_exc_info_with_unicode():
+    exceptions_to_raise = [Exception(u"DataDog/水")]
+    if six.PY3:
+        exceptions_to_raise.append(Exception("DataDog/水"))
```

Comment:
I think this was fine before. In Python 3 `u"水" == "水"` evaluates to `True` so those exceptions are effectively the same.

Reply:
mmm using `ensure_text` make some other tests fail 🤔 . I'm really confused at this point 🤣

Reply:
@majorgreys WDYT? How did we get out of similar situations in the past?

Reply:
for example: https://github.com/DataDog/dd-trace-py/runs/4289758575?check_suite_focus=true 
```
  File "/opt/hostedtoolcache/Python/3.9.8/x64/lib/python3.9/site-packages/six.py", line 955, in ensure_text
    raise TypeError("not expecting type '%s'" % type(s))
TypeError: not expecting type '<class 'ValueError'>'
```

Reply:
yes, that's because `ensure_text` expects a string-like object, not a subtype of `Exception`. It looks like serialising exceptions is not trivial in Python 2. If the message associated with the exception is enough, then a possible way out is `ensure_text(exc_value.message)`.

Reply:
I _think_ it should be fine